cocos/renderer/gfx-metal/MTLCommandBuffer.mm

```diff
@@ -166,8 +167,15 @@ of this software and associated engine source code (the "Software"), a limited,
     else {
         _renderEncoder.initialize(_mtlCommandBuffer, mtlRenderPassDescriptor);
     }
-    _renderEncoder.setViewport(renderArea);
-    _renderEncoder.setScissor(renderArea);
+
+    Rect       scissorArea      = renderArea;
+#if defined(CC_DEBUG) && (CC_DEBUG > 0)
```

Comment:
only debug mode there is a validation layer, renderArea is not consistent with rendertarget.w/h sometimes.
nativeWindow::resize->cc::EventDispatcher::resize->(...jsb call...)->js set camera width->(...jsb call...)->native beginrenderpass get renderArea. this long process makes a time delay, but rendertarget comes from drawable.texture directly, it's real time, thus inconsistent.

Reply:
What will happen if they are consistent in release mode?

Reply:
API validation is ignored, so no scissor area validation.

Reply:
The effect is right?

Reply:
yes, just ignore some assert, no other influence